scripts/zkai_snarkjs_statement_envelope_benchmark.py

```
@@ -0,0 +1,591 @@
+def _case_result(
+    adapter: str,
+    envelope: dict[str, Any],
+    external_verify: Callable[[dict[str, Any], list[Any], dict[str, Any]], None],
+) -> tuple[bool, str]:
+    try:
+        if adapter == "snarkjs-proof-only":
+            verify_proof_only(envelope, external_verify=external_verify)
+        elif adapter == "snarkjs-statement-envelope":
+            verify_statement_envelope(envelope, external_verify=external_verify)
+        else:
+            raise SnarkjsEnvelopeError(f"unsupported adapter {adapter!r}")
```

Ruff (0.15.12)

[warning] Abstract `raise` to an inner function

(TRY301)